### ISSUE TITLE
db.createRole doesn't work

### ISSUE THREAD

**Issue by spacecat** [mongodb/mongodb-kubernetes-operator#274]
**What did you do to encounter the bug?**
1. Create CR

```
---
apiVersion: mongodb.com/v1
kind: MongoDB
metadata:
  name: mongodb
spec:
  members: 3
  type: ReplicaSet
  version: "4.2.6"
  security:
    authentication:
      modes: ["SCRAM"]
  users:
    - name: admin
      db: admin
      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
        name: my-user-password
      roles:
        - name: root
          db: admin
      scramCredentialsSecretName: my-scram
  statefulSet:
    spec:
      template:
        spec:
          containers:
            - name: "mongodb-agent"
              resources:
                requests:
                  cpu: 50m
                  memory: 250M
                limits:
                  cpu: 100m
                  memory: 500M
            - name: "mongod"
              resources:
                requests:
                  cpu: 100m
                  memory: 1000M
                limits:
                  cpu: 200m
                  memory: 2000M

# the user credentials will be generated from this secret
# once the credentials are generated, this secret is no longer required
---
apiVersion: v1
kind: Secret
metadata:
  name: my-user-password
type: Opaque
stringData:
  password: blah

```

2. Log in to mongo

`k exec --stdin --tty mongodb-0 -c mongod -- /bin/bash`

`mongo "mongodb://mongodb-svc.webv4.svc.cluster.local:27017/?replicaSet=mongodb" --username admin --password blah --authenticationDatabase admin`

or just `mongo`.

3. Run the following command:

`db.createRole({ role: "restoreOperator", privileges: [{ resource: { anyResource: true }, actions: [ "anyAction" ] }], roles: [] })`

**What did you expect?**
`db.createRole` should create the role successfully.

**What happened instead?**
Okay, here comes the crazy part.

When I run the `db.createRole` command, it executes successfully.

I then run `db.getRoles({ rolesInfo: 1, showPrivileges:true, showBuiltinRoles: false })` to verify this and the role is indeed in the array returned by `db.getRoles`.

But then; I wait 2-3 seconds, and I run `db.getRoles({ rolesInfo: 1, showPrivileges:true, showBuiltinRoles: false })` again and `restoreOperator` is not there anymore!

I know the role gets created because if I run `db.getRoles` immediately after `db.createRole` I get the following output:

```
mongodb:PRIMARY> db.getRoles({ rolesInfo: 1, showPrivileges:true, showBuiltinRoles: false })
[
        {
                "role" : "restoreOperator",
                "db" : "admin",
                "isBuiltin" : false,
                "roles" : [ ],
                "inheritedRoles" : [ ],
                "privileges" : [
                        {
                                "resource" : {
                                        "anyResource" : true
                                },
                                "actions" : [
                                        "anyAction"
                                ]
                        }
                ],
                "inheritedPrivileges" : [
                        {
                                "resource" : {
                                        "anyResource" : true
                                },
                                "actions" : [
                                        "anyAction"
                                ]
                        }
                ]
        }
]
```
And if I try to add the role again I get the following error message:

```
mongodb:PRIMARY> db.createRole({ role: "restoreOperator", privileges: [{ resource: { anyResource: true }, actions: [ "anyAction" ] }], roles: [] })
2021-01-07T15:03:09.613+0000 E  QUERY    [js] uncaught exception: Error: Role "restoreOperator@admin" already exists :
_getErrorWithCode@src/mongo/shell/utils.js:25:13
DB.prototype.createRole@src/mongo/shell/db.js:1654:15
@(shell):1:1
```

🤯

**Operator Information**
 - Operator Version
 ```
❯ k describe deploy mongodb-kubernetes-operator | grep Image
    Image:      quay.io/mongodb/mongodb-kubernetes-operator:0.3.0
```
 - MongoDB Image used
```
❯ k describe sts mongodb | grep Image
    Image:      quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2
    Image:      quay.io/mongodb/mongodb-agent:10.19.0.6562-1
    Image:      registry.hub.docker.com/library/mongo:4.2.6
```

**Kubernetes Cluster Information**
 - Distribution
 Azure AKS
 - Version
 1.19.3
 - Image Registry location (quay, or an internal registry)
 quay

**Additional context**
The reason I want to be able to create a user-defined role with these privileges is because I need to run mongorestore.

 I need to run mongorestore with the --oplogReplay argument flag. This is required if you've used mongodump with the --oplog flag.

The docs clearly states this:

https://docs.mongodb.com/database-tools/mongorestore/#bin.mongorestore

--oplogReplay | To run with --oplogReplay, create a user-defined role that has anyAction on anyResource.

Also, none of the built-in roles have these privileges so I _must_ create my own role.

```
Every 2.0s: kubectl get all                                           Thu Jan  7 16:42:15 2021
NAME                                               READY   STATUS    RESTARTS   AGE
pod/mongodb-0                                      2/2     Running   0          81m
pod/mongodb-1                                      2/2     Running   0          78m
pod/mongodb-2                                      2/2     Running   0          76m
pod/mongodb-kubernetes-operator-7d557bcc95-p7j7d   1/1     Running   0          81m

NAME                  TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)     AGE
service/mongodb-svc   ClusterIP   None         <none>        27017/TCP   20h

NAME                                          READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/mongodb-kubernetes-operator   1/1     1            1           22h

NAME                                                     DESIRED   CURRENT   READY   AGE
replicaset.apps/mongodb-kubernetes-operator-7d557bcc95   1         1         1       22h

NAME                       READY   AGE
statefulset.apps/mongodb   3/3     20h

```

```

❯ k describe sts mongodb
Name:               mongodb
Namespace:          webv4
CreationTimestamp:  Wed, 06 Jan 2021 20:22:50 +0100
Selector:           app=mongodb-svc
Labels:             app=mongodb-svc
Annotations:        <none>
Replicas:           3 desired | 3 total
Update Strategy:    RollingUpdate
Pods Status:        3 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           app=mongodb-svc
  Service Account:  mongodb-kubernetes-operator
  Init Containers:
   mongod-posthook:
    Image:      quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2
    Port:       <none>
    Host Port:  <none>
    Command:
      cp
      version-upgrade-hook
      /hooks/version-upgrade
    Environment:  <none>
    Mounts:
      /hooks from hooks (rw)
  Containers:
   mongodb-agent:
    Image:      quay.io/mongodb/mongodb-agent:10.19.0.6562-1
    Port:       <none>
    Host Port:  <none>
    Command:
      agent/mongodb-agent
      -cluster=/var/lib/automation/config/cluster-config.json
      -skipMongoStart
      -noDaemonize
      -healthCheckFilePath=/var/log/mongodb-mms-automation/healthstatus/agent-health-status.json
      -serveStatusPort=5000
      -useLocalMongoDbTools
    Limits:
      cpu:     100m
      memory:  500M
    Requests:
      cpu:      50m
      memory:   250M
    Readiness:  exec [/var/lib/mongodb-mms-automation/probes/readinessprobe] delay=5s timeout=1s period=10s #success=1 #failure=60
    Environment:
      AGENT_STATUS_FILEPATH:  /var/log/mongodb-mms-automation/healthstatus/agent-health-status.json
      AUTOMATION_CONFIG_MAP:  mongodb-config
      HEADLESS_AGENT:         true
      POD_NAMESPACE:           (v1:metadata.namespace)
    Mounts:
      /data from data-volume (rw)
      /var/lib/automation/config from automation-config (ro)
      /var/lib/mongodb-mms-automation/authentication from mongodb-agent-scram-credentials (rw)
      /var/log/mongodb-mms-automation/healthstatus from healthstatus (rw)
   mongod:
    Image:      registry.hub.docker.com/library/mongo:4.2.6
    Port:       <none>
    Host Port:  <none>
    Command:
      /bin/sh
      -c

      # run post-start hook to handle version changes
      /hooks/version-upgrade

      # wait for config to be created by the agent
      while [ ! -f /data/automation-mongod.conf ]; do sleep 3 ; done ; sleep 2 ;

      # start mongod with this configuration
      exec mongod -f /data/automation-mongod.conf ;

    Limits:
      cpu:     200m
      memory:  2G
    Requests:
      cpu:     100m
      memory:  1G
    Environment:
      AGENT_STATUS_FILEPATH:  /healthstatus/agent-health-status.json
    Mounts:
      /data from data-volume (rw)
      /healthstatus from healthstatus (rw)
      /hooks from hooks (rw)
      /var/lib/mongodb-mms-automation/authentication from mongodb-agent-scram-credentials (rw)
  Volumes:
   automation-config:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mongodb-config
    Optional:    false
   healthstatus:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
   hooks:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
   mongodb-agent-scram-credentials:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mongodb-agent-scram-credentials
    Optional:    false
Volume Claims:
  Name:          data-volume
  StorageClass:
  Labels:        <none>
  Annotations:   <none>
  Capacity:      10G
  Access Modes:  [ReadWriteOnce]
Events:          <none>
``` 
```

❯ k describe replicaset mongodb-kubernetes-operator-7d557bcc95
Name:           mongodb-kubernetes-operator-7d557bcc95
Namespace:      webv4
Selector:       name=mongodb-kubernetes-operator,pod-template-hash=7d557bcc95
Labels:         name=mongodb-kubernetes-operator
                pod-template-hash=7d557bcc95
Annotations:    deployment.kubernetes.io/desired-replicas: 1
                deployment.kubernetes.io/max-replicas: 2
                deployment.kubernetes.io/revision: 1
Controlled By:  Deployment/mongodb-kubernetes-operator
Replicas:       1 current / 1 desired
Pods Status:    1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:           name=mongodb-kubernetes-operator
                    pod-template-hash=7d557bcc95
  Service Account:  mongodb-kubernetes-operator
  Containers:
   mongodb-kubernetes-operator:
    Image:      quay.io/mongodb/mongodb-kubernetes-operator:0.3.0
    Port:       <none>
    Host Port:  <none>
    Command:
      mongodb-kubernetes-operator
    Environment:
      WATCH_NAMESPACE:              (v1:metadata.namespace)
      POD_NAME:                     (v1:metadata.name)
      OPERATOR_NAME:               mongodb-kubernetes-operator
      AGENT_IMAGE:                 quay.io/mongodb/mongodb-agent:10.19.0.6562-1
      VERSION_UPGRADE_HOOK_IMAGE:  quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2
      MONGODB_IMAGE:               library/mongo
      MONGODB_REPO_URL:            registry.hub.docker.com
    Mounts:                        <none>
  Volumes:                         <none>
Events:                            <none>
```

```

❯ k describe deploy mongodb-kubernetes-operator
Name:                   mongodb-kubernetes-operator
Namespace:              webv4
CreationTimestamp:      Wed, 06 Jan 2021 18:10:45 +0100
Labels:                 <none>
Annotations:            deployment.kubernetes.io/revision: 1
Selector:               name=mongodb-kubernetes-operator
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           name=mongodb-kubernetes-operator
  Service Account:  mongodb-kubernetes-operator
  Containers:
   mongodb-kubernetes-operator:
    Image:      quay.io/mongodb/mongodb-kubernetes-operator:0.3.0
    Port:       <none>
    Host Port:  <none>
    Command:
      mongodb-kubernetes-operator
    Environment:
      WATCH_NAMESPACE:              (v1:metadata.namespace)
      POD_NAME:                     (v1:metadata.name)
      OPERATOR_NAME:               mongodb-kubernetes-operator
      AGENT_IMAGE:                 quay.io/mongodb/mongodb-agent:10.19.0.6562-1
      VERSION_UPGRADE_HOOK_IMAGE:  quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2
      MONGODB_IMAGE:               library/mongo
      MONGODB_REPO_URL:            registry.hub.docker.com
    Mounts:                        <none>
  Volumes:                         <none>
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Progressing    True    NewReplicaSetAvailable
  Available      True    MinimumReplicasAvailable
OldReplicaSets:  <none>
NewReplicaSet:   mongodb-kubernetes-operator-7d557bcc95 (1/1 replicas created)
Events:          <none>

```

```

❯ k describe svc mongodb-svc
Name:              mongodb-svc
Namespace:         webv4
Labels:            <none>
Annotations:       <none>
Selector:          app=mongodb-svc
Type:              ClusterIP
IP Families:       <none>
IP:                None
IPs:               <none>
Port:              <unset>  27017/TCP
TargetPort:        27017/TCP
Endpoints:         10.244.0.5:27017,10.244.1.3:27017,10.244.2.4:27017
Session Affinity:  None
Events:            <none>


```

```

❯ k describe po mongodb-kubernetes-operator-7d557bcc95-p7j7d
Name:         mongodb-kubernetes-operator-7d557bcc95-p7j7d
Namespace:    webv4
Priority:     0
Node:         aks-agentpool-85856963-vmss000003/10.240.0.4
Start Time:   Thu, 07 Jan 2021 15:21:22 +0100
Labels:       name=mongodb-kubernetes-operator
              pod-template-hash=7d557bcc95
Annotations:  <none>
Status:       Running
IP:           10.244.0.3
IPs:
  IP:           10.244.0.3
Controlled By:  ReplicaSet/mongodb-kubernetes-operator-7d557bcc95
Containers:
  mongodb-kubernetes-operator:
    Container ID:  containerd://26a687cd539a41a8f77650bc51cdda87aa64cd69d27d5e1efc8dad578ba765bf
    Image:         quay.io/mongodb/mongodb-kubernetes-operator:0.3.0
    Image ID:      quay.io/mongodb/mongodb-kubernetes-operator@sha256:107a7c73af5953c473060ee84a5a0b8e690776f384486701223b63993dea1514
    Port:          <none>
    Host Port:     <none>
    Command:
      mongodb-kubernetes-operator
    State:          Running
      Started:      Thu, 07 Jan 2021 15:21:46 +0100
    Ready:          True
    Restart Count:  0
    Environment:
      WATCH_NAMESPACE:             webv4 (v1:metadata.namespace)
      POD_NAME:                    mongodb-kubernetes-operator-7d557bcc95-p7j7d (v1:metadata.name)
      OPERATOR_NAME:               mongodb-kubernetes-operator
      AGENT_IMAGE:                 quay.io/mongodb/mongodb-agent:10.19.0.6562-1
      VERSION_UPGRADE_HOOK_IMAGE:  quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2
      MONGODB_IMAGE:               library/mongo
      MONGODB_REPO_URL:            registry.hub.docker.com
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from mongodb-kubernetes-operator-token-gk575 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  mongodb-kubernetes-operator-token-gk575:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mongodb-kubernetes-operator-token-gk575
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:          <none>
```

```
❯ k describe po mongodb-0
Name:         mongodb-0
Namespace:    webv4
Priority:     0
Node:         aks-agentpool-85856963-vmss000003/10.240.0.4
Start Time:   Thu, 07 Jan 2021 15:21:22 +0100
Labels:       app=mongodb-svc
              controller-revision-hash=mongodb-7d9d5888c6
              statefulset.kubernetes.io/pod-name=mongodb-0
Annotations:  <none>
Status:       Running
IP:           10.244.0.5
IPs:
  IP:           10.244.0.5
Controlled By:  StatefulSet/mongodb
Init Containers:
  mongod-posthook:
    Container ID:  containerd://787c08c61d548e3e2f0e0faf8634b70c27c0048e287f4ad53af75c675556e8ae
    Image:         quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2
    Image ID:      quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook@sha256:e99105b1c54e12913ddaf470af8025111a6e6e4c8917fc61be71d1bc0328e7d7
    Port:          <none>
    Host Port:     <none>
    Command:
      cp
      version-upgrade-hook
      /hooks/version-upgrade
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Thu, 07 Jan 2021 15:21:51 +0100
      Finished:     Thu, 07 Jan 2021 15:21:51 +0100
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /hooks from hooks (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from mongodb-kubernetes-operator-token-gk575 (ro)
Containers:
  mongodb-agent:
    Container ID:  containerd://5f5747fe242bc417df9f1e3d22f1a03b399de1b0ea2bf4d4153965cf5a56723d
    Image:         quay.io/mongodb/mongodb-agent:10.19.0.6562-1
    Image ID:      quay.io/mongodb/mongodb-agent@sha256:790c2670ef7cefd61cfaabaf739de16dbd2e07dc3b539add0da21ab7d5ac7626
    Port:          <none>
    Host Port:     <none>
    Command:
      agent/mongodb-agent
      -cluster=/var/lib/automation/config/cluster-config.json
      -skipMongoStart
      -noDaemonize
      -healthCheckFilePath=/var/log/mongodb-mms-automation/healthstatus/agent-health-status.json
      -serveStatusPort=5000
      -useLocalMongoDbTools
    State:          Running
      Started:      Thu, 07 Jan 2021 15:22:36 +0100
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  500M
    Requests:
      cpu:      50m
      memory:   250M
    Readiness:  exec [/var/lib/mongodb-mms-automation/probes/readinessprobe] delay=5s timeout=1s period=10s #success=1 #failure=60
    Environment:
      AGENT_STATUS_FILEPATH:  /var/log/mongodb-mms-automation/healthstatus/agent-health-status.json
      AUTOMATION_CONFIG_MAP:  mongodb-config
      HEADLESS_AGENT:         true
      POD_NAMESPACE:          webv4 (v1:metadata.namespace)
    Mounts:
      /data from data-volume (rw)
      /var/lib/automation/config from automation-config (ro)
      /var/lib/mongodb-mms-automation/authentication from mongodb-agent-scram-credentials (rw)
      /var/log/mongodb-mms-automation/healthstatus from healthstatus (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from mongodb-kubernetes-operator-token-gk575 (ro)
  mongod:
    Container ID:  containerd://ab2e010d1665a2c50e468a99c44b7c5e5d8b10c94633aff8ca7999b1e05bca22
    Image:         registry.hub.docker.com/library/mongo:4.2.6
    Image ID:      registry.hub.docker.com/library/mongo@sha256:c880f6b56f443bb4d01baa759883228cd84fa8d78fa1a36001d1c0a0712b5a07
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/sh
      -c

      # run post-start hook to handle version changes
      /hooks/version-upgrade

      # wait for config to be created by the agent
      while [ ! -f /data/automation-mongod.conf ]; do sleep 3 ; done ; sleep 2 ;

      # start mongod with this configuration
      exec mongod -f /data/automation-mongod.conf ;

    State:          Running
      Started:      Thu, 07 Jan 2021 15:23:10 +0100
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     200m
      memory:  2G
    Requests:
      cpu:     100m
      memory:  1G
    Environment:
      AGENT_STATUS_FILEPATH:  /healthstatus/agent-health-status.json
    Mounts:
      /data from data-volume (rw)
      /healthstatus from healthstatus (rw)
      /hooks from hooks (rw)
      /var/lib/mongodb-mms-automation/authentication from mongodb-agent-scram-credentials (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from mongodb-kubernetes-operator-token-gk575 (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  data-volume:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  data-volume-mongodb-0
    ReadOnly:   false
  automation-config:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mongodb-config
    Optional:    false
  healthstatus:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  hooks:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  mongodb-agent-scram-credentials:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mongodb-agent-scram-credentials
    Optional:    false
  mongodb-kubernetes-operator-token-gk575:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mongodb-kubernetes-operator-token-gk575
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:          <none>



```

Update 1: 

I added:

roles:
```
- name: __system
          db: admin
```

to my CR and ran mongorestore again but it still fails:

`[info]: .\mongorestore                : Failed: restore error: error applying oplog: applyOps: (Unauthorized) command applyOps requires authentication`

Also; I'm getting the following errors when running mongorestore:
```

[info]: .\mongorestore                : don't know what to do with subdirectory "SquidexContent", skipping...
[info]: .\mongorestore                : don't know what to do with subdirectory "Squidex", skipping...
[info]: .\mongorestore                : don't know what to do with subdirectory "SquidexAssets", skipping...
[info]: .\mongorestore                : don't know what to do with subdirectory "", skipping...
[info]: .\mongorestore                : don't know what to do with subdirectory "admin", skipping...
```

Anyone know what it means?

**Comment by antonlisovenko** [comment]
Hi @spacecat 

Thanks for your report.
The magical behavior you encounter is caused by the Agent that overrides any custom roles. All the roles are supposed to be configured from the Kubernetes MongoDB Custom Resource only, but so far only the set of predefined roles is created. Custom roles are not implemented yet.

We are planning to add the support for custom roles - tracking it in the internal ticket.

**Comment by spacecat** [comment]
Is there any work-around at the moment?

**Comment by antonlisovenko** [comment]
I'm afraid no. The agent is explicitly configured to override the custom roles and we don't plan to revert this considering that custom roles will be supported from CRD

**Comment by spacecat** [comment]
Thanks @antonlisovenko 
Is there an ETA for the custom roles implementation?

**Comment by spacecat** [comment]
I see this has been implemented in v0.4.0. Great. Do the docs reflect these changes as well? Are there any examples for example?

**Comment by alyacb** [comment]
Hi @spacecat!

The docs have been updated to include a section on [how to define custom roles](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/deploy-configure.md#define-a-custom-database-role) and there is an example cr in [deploy/crds/mongodb.com_v1_mongodb_custom_role.yaml](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/deploy/crds/mongodb.com_v1_mongodb_custom_role.yaml).
